bndtools.core/src/org/bndtools/core/editors/ImportPackageQuickFixProcessor.java

```diff
@@ -0,0 +1,520 @@
+import org.eclipse.core.resources.IResourceRuleFactory;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.resources.WorkspaceJob;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.compiler.IProblem;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ImportDeclaration;
+import org.eclipse.jdt.core.dom.Name;
+import org.eclipse.jdt.core.dom.NameQualifiedType;
+import org.eclipse.jdt.core.dom.PackageDeclaration;
+import org.eclipse.jdt.core.dom.QualifiedName;
+import org.eclipse.jdt.core.dom.SimpleType;
+import org.eclipse.jdt.internal.corext.dom.ASTNodes;
+import org.eclipse.jdt.ui.text.java.IInvocationContext;
+import org.eclipse.jdt.ui.text.java.IJavaCompletionProposal;
+import org.eclipse.jdt.ui.text.java.IProblemLocation;
+import org.eclipse.jdt.ui.text.java.IQuickFixProcessor;
+import org.eclipse.jface.text.contentassist.IContextInformation;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.graphics.Point;
+import org.osgi.framework.namespace.BundleNamespace;
+import org.osgi.framework.namespace.PackageNamespace;
+import org.osgi.resource.Capability;
+import org.osgi.resource.Requirement;
+import org.osgi.service.repository.Repository;
+
+import aQute.bnd.build.Project;
+import aQute.bnd.build.Workspace;
+import aQute.bnd.build.WorkspaceRepository;
+import aQute.bnd.build.model.BndEditModel;
+import aQute.bnd.build.model.clauses.VersionedClause;
+import aQute.bnd.header.Attrs;
+import aQute.bnd.osgi.resource.CapReqBuilder;
+import aQute.bnd.properties.Document;
+import aQute.bnd.properties.IDocument;
+import aQute.bnd.service.RepositoryPlugin;
+import aQute.lib.collections.MultiMap;
+import aQute.lib.io.ByteBufferInputStream;
+import aQute.lib.io.IO;
+import aQute.lib.strings.Strings;
+import bndtools.central.Central;
+import bndtools.central.RepositoryUtils;
+
+public class ImportPackageQuickFixProcessor implements IQuickFixProcessor {
+
+    private static final ILogger logger = Logger.getLogger(ImportPackageQuickFixProcessor.class);
+
+    // Relevance constants
+    final public static int ADD_BUNDLE = 15;
+    final public static int ADD_BUNDLE_WORKSPACE = ADD_BUNDLE + 1;
+
+    @Override
+    public boolean hasCorrections(ICompilationUnit unit, int problemId) {
+
+        switch (problemId) {
+            case IProblem.ImportNotFound :
+            case IProblem.UndefinedType :
+                return true;
+            default :
+                return false;
+        }
+    }
+
+    static private Requirement buildRequirement(String importName) {
+        String filter = String.format("(%s=%s)", PackageNamespace.PACKAGE_NAMESPACE, importName.trim());
```

Comment:
I could be wrong, but I feel that we could use one of the `add...()` methods on `CapReqBuilder` to handle this bit. If I'm wrong feel free to ignore.

Reply:
I have managed to change this to use `CapReqBuilder` exclusively, as you suggested.

Note that the implementation of `buildRequirement()` was originally lifted from here: https://github.com/bndtools/bndtools/blob/ca95e3ca7ba2344ec21275eea29b166d1dbc7fcc/bndtools.core/src/bndtools/views/repository/PackageSearchPanel.java#L94-L97

and here:
https://github.com/bndtools/bndtools/blob/ca95e3ca7ba2344ec21275eea29b166d1dbc7fcc/bndtools.core/src/bndtools/views/repository/PackageSearchPanel.java#L105-L110

... so maybe the search code in `bndtools.views.repository` could also be simplified in this way.